modules/renter/contractor/persist.go

```diff
@@ -80,6 +75,17 @@ func (c *Contractor) load() error {
 		}
 		c.currentPeriod = highestEnd - c.allowance.Period
 	}
+
+	// Old perist may need information from the hostdb to fill in missing
+	// fields.
+	allHosts := c.hdb.AllHosts()
+	hmap := make(map[modules.NetAddress]modules.HostDBEntry)
+	// Iterate so that in the event of duplicate hosts for a netaddress, the
+	// highest score host is the one in the map.
+	for _, host := range allHosts {
+		hmap[host.NetAddress] = host
+	}
```

Comment:
This is almost certainly insufficient, as netaddresses that change are dropped from the hostdb altogether - if your host changed address since you formed the contract, it's game over :(

I don't think there's an easy way to deal with this, unless we are scanning the whole blockchain or keeping a only-used-once entire extra field in the hostdb.

Might just be a scar. It's strictly not worse than what is already deployed, so at least it's not a regression =\